src/routes/stake/Stake.svelte

```diff
@@ -35,7 +35,7 @@
 	let totalIcpDeposited: BigNumber;
 
 	async function icpToNicp(amount: BigNumber) {
-		if (!$user || $isConverting || !$canisters || amount.isNaN() || amount.isLessThan(BigNumber(1)))
+		if (!$user || $isConverting || !$canisters || !$canisters.waterNeuron.authenticatedActor || amount.isNaN() || amount.isLessThan(BigNumber(1)))
```

Comment:
isn't $canisters always set now?

Reply:
The same way as it was before. The thing is, I instantiate the canisters in the onMount of the `+layout.svelte` file but depending on how the DOM renders, it might not be defined when rendering other components. To avoid the we use `Canisters | undefined`

Reply:
Okay, not sure if it's the best, we should maybe ask to David or Nicolas what they think.